repl/src/main/scala/com/cloudera/livy/repl/PythonInterpreter.scala

```diff
@@ -192,5 +243,32 @@ private class PythonInterpreter(process: Process, gatewayServer: GatewayServer,
       parse(line)
     }
   }
+
+  def createWrapper(driver: ReplDriver, msg: BaseProtocol.BypassJobRequest): BypassJobWrapper = {
+    new BypassJobWrapper(driver, msg.id,
+      new BypassPySparkJob(msg.serializedJob, pysparkJobProcessor))
+  }
+
+  def addFile(path: String): Unit = {
+    pysparkJobProcessor.addFile(path)
+  }
+
+  def addPyFile(driver: ReplDriver, conf: SparkConf, path: String): Unit = {
+    val localCopyDir = new File(pysparkJobProcessor.getLocalTmpDirPath)
+    val localCopyFile = driver.copyFileToLocal(localCopyDir, path, SparkContext.getOrCreate(conf))
+    pysparkJobProcessor.addPyFile(localCopyFile.getPath)
+    if (path.endsWith(".jar")) {
+      driver.addLocalFileToClassLoader(localCopyFile)
+    }
+  }
+
+  private def updatePythonGatewayPort(port: Int): Unit = {
+    if (port != 0) {
```

Comment:
The reason for my question was that if this cannot be 0, you don't need the check. If it should not be 0, but for some reason can be 0, then you need to handle it as an error.